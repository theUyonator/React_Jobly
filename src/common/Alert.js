import React from "react";

/**This component shows bootstrap style alerts  */

function Alert({ type = "danger", messages = []}){
    return (
        <div className={`alert alert-${type}`} role="alert">
            {messages.map(err => (
                <p className="mb-0 small" key={err}>
                    {err}
                </p>
            ))}
        </div>
    );

}

export default Alert;