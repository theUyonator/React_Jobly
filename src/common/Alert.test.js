import React from "react";
import { render } from "@testing-library/react";
import Alert from "./Alert";

test("renders without crashing", function () {
    render(<Alert />);
});

test("matches snapshot for danger", function () {
    let messages = ["Danger everywhere", "But go to see it through ma boi!"];
    const { asFragment } = render(<Alert type="danger" messages={messages} />);
    expect(asFragment()).toMatchSnapshot();
});

test("matches snapshot for success", function () {
    let messages = ["Everythang good over here boi!", "Thumbs up ma boi!"];
    const { asFragment } = render(<Alert type="success" messages={messages} />);
    expect(asFragment()).toMatchSnapshot();
});