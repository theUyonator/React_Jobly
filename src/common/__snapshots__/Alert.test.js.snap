// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for danger 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      Danger everywhere
    </p>
    <p
      class="mb-0 small"
    >
      But go to see it through ma boi!
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      Everythang good over here boi!
    </p>
    <p
      class="mb-0 small"
    >
      Thumbs up ma boi!
    </p>
  </div>
</DocumentFragment>
`;
